haystack/components/routers/conditional_router.py

```diff
@@ -166,7 +205,26 @@ def __init__(
             # extract outputs
             output_types.update({route["output_name"]: route["output_type"]})
 
-        component.set_input_types(self, **{var: Any for var in input_types})
+        # remove optional variables from mandatory input types
+        mandatory_input_types = input_types - set(self.optional_variables)
+
+        # warn about unused optional variables
+        unused_optional_vars = set(self.optional_variables) - input_types if self.optional_variables else None
+        if unused_optional_vars:
+            warn(
+                f"The following optional variables are specified but not used in any route: {unused_optional_vars}. "
+                "Check if there's a typo in variable names.",
+                UserWarning,
+            )
```

Comment:
Could we use the `logger.warning` here instead or in addition to? Using the logger allows us to surface these warnings to users in dC more easily.